Maven upload with pathes does not work

### Description

Hello there ☕ 
as the S3 issue is now fixed i wanted to test hosted and virtual maven repositories. I created a new hosted repository first and a dummy spring application. afterwards i wanted to run mvn deploy but it fails to verify that the jar and pom have been uploaded. Manual uploading the jar via the web ui with path given to the same value as the package, results in the jar being uploaded on root level and not in the given path, so i believe a bug there, or i am too stupid to set it up correctly.

### Environment Information

- Deployment type: k8s
- Operating System: alpine
- Architecture: amd64
- Database: postgresql 
- Reverse Proxy: nginx ingress
- Network setup: cilium


### Backend Version

ghcr.io/artifact-keeper/artifact-keeper-backend@sha256:21b017ff615986b7b5b953f8ad6ed5ea4a5f69afc07ae77c91d69065df3091f2

### Web UI Version

1.1.0-rc.8

### Package Format

Maven / Gradle

### Expected Behavior

The upload should succeed and the given path (package) should be parsed correctly and be displayed in the ui

### Observed Behavior

mvn deploy fails with 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.1.4:deploy (default-deploy) on project artifacthub: Failed to deploy artifacts: Could not find artifact com.test:artifacthub:pom:0.0.1-20260304.095300-1 in gradle-test (https://REDACTED/maven/repo-test/) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.1.4:deploy (default-deploy) on project artifacthub: Failed to deploy artifacts: Could not find artifact com.test:artifacthub:pom:0.0.1-20260304.095300-1 in gradle-test (https://REDACTED/maven/repo-test/)
```

### Steps to Reproduce

create new hosted repository
create dummy project with https://start.spring.io/
configure auth in ~/.m2 with newly created token
configure distributionManagement in pom.xml
run `./mvnw deploy`

### Logs

```shell
No relevant logs found
```

### Additional Context

pom.xml snippet: 

```xml
	<distributionManagement>
		<repository>
			<id>repo-test</id>
			<url>https://REDACTED/maven/repo-test/</url>
		</repository>
		<snapshotRepository>
			<id>repo-test</id>
			<url>https://REDACTED/maven/repo-test/</url>
		</snapshotRepository>
	</distributionManagement>
```

~/.m2/settings.xml

```xml
<?xml version="1.0"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                                                http://maven.apache.org/xsd/settings-1.0.0.xsd">

    <servers>
    <server>
      <id>repo-test</id>
      <username>admin</username>
      <password>REDACTED</password>
    </server>
  </servers>
</settings>
```

### Pre-submission Checklist

- [x] I have searched existing issues to ensure this isn't a duplicate
- [x] I have redacted any sensitive information (passwords, tokens, private URLs)
- [x] I am using a supported version (not end-of-life)

--- a/backend/src/api/handlers/maven.rs
+++ b/backend/src/api/handlers/maven.rs
@@ -165,38 +165,80 @@ async fn download(
     Path((repo_key, path)): Path<(String, String)>,
 ) -> Result<Response, Response> {
     let repo = resolve_maven_repo(&state.db, &repo_key).await?;
+    let storage = state.storage_for_repo(&repo.storage_path);
 
     // 1. Check if this is a checksum request for metadata
     if let Some((base_path, checksum_type)) = parse_checksum_path(&path) {
-        if let Some((group_id, artifact_id)) = parse_metadata_path(base_path) {
-            let xml =
-                generate_metadata_for_artifact(&state.db, repo.id, &group_id, &artifact_id).await?;
-            let checksum = compute_checksum(xml.as_bytes(), checksum_type);
-            return Ok(Response::builder()
-                .status(StatusCode::OK)
-                .header(CONTENT_TYPE, "text/plain")
-                .body(Body::from(checksum))
-                .unwrap());
+        if MavenHandler::is_metadata(base_path) {
+            // Try stored checksum file first
+            let checksum_storage_key = format!("maven/{}", path);
+            if let Ok(content) = storage.get(&checksum_storage_key).await {
+                return Ok(Response::builder()
+                    .status(StatusCode::OK)
+                    .header(CONTENT_TYPE, "text/plain")
+                    .body(Body::from(content))
+                    .unwrap());
+            }
+
+            // Try stored metadata file and compute checksum from it
+            let meta_storage_key = format!("maven/{}", base_path);
+            if let Ok(content) = storage.get(&meta_storage_key).await {
+                let checksum = compute_checksum(&content, checksum_type);
+                return Ok(Response::builder()
+                    .status(StatusCode::OK)
+                    .header(CONTENT_TYPE, "text/plain")
+                    .body(Body::from(checksum))
+                    .unwrap());
+            }
+
+            // Fall back to dynamic generation for artifact-level metadata
+            if let Some((group_id, artifact_id)) = parse_metadata_path(base_path) {
+                let xml =
+                    generate_metadata_for_artifact(&state.db, repo.id, &group_id, &artifact_id)
+                        .await?;
+                let checksum = compute_checksum(xml.as_bytes(), checksum_type);
+                return Ok(Response::builder()
+                    .status(StatusCode::OK)
+                    .header(CONTENT_TYPE, "text/plain")
+                    .body(Body::from(checksum))
+                    .unwrap());
+            }
         }
     }
 
     // 2. Check if this is a maven-metadata.xml request
-    if let Some((group_id, artifact_id)) = parse_metadata_path(&path) {
-        let xml =
-            generate_metadata_for_artifact(&state.db, repo.id, &group_id, &artifact_id).await?;
-        return Ok(Response::builder()
-            .status(StatusCode::OK)
-            .header(CONTENT_TYPE, "application/xml")
-            .header(CONTENT_LENGTH, xml.len().to_string())
-            .body(Body::from(xml))
-            .unwrap());
+    if MavenHandler::is_metadata(&path) {
+        // Try stored metadata file first (handles version-level metadata)
+        let meta_storage_key = format!("maven/{}", path);
+        if let Ok(content) = storage.get(&meta_storage_key).await {
+            return Ok(Response::builder()
+                .status(StatusCode::OK)
+                .header(CONTENT_TYPE, "application/xml")
+                .header(CONTENT_LENGTH, content.len().to_string())
+                .body(Body::from(content))
+                .unwrap());
+        }
+
+        // Fall back to dynamic generation for artifact-level metadata
+        if let Some((group_id, artifact_id)) = parse_metadata_path(&path) {
+            let xml =
+                generate_metadata_for_artifact(&state.db, repo.id, &group_id, &artifact_id).await?;
+            return Ok(Response::builder()
+                .status(StatusCode::OK)
+                .header(CONTENT_TYPE, "application/xml")
+                .header(CONTENT_LENGTH, xml.len().to_string())
+                .body(Body::from(xml))
+                .unwrap());
+        }
+
+        // Metadata not found anywhere
+        return Err((StatusCode::NOT_FOUND, "Metadata not found").into_response());
     }
 
     // 3. Check if this is a checksum request for a stored file
     if let Some((base_path, checksum_type)) = parse_checksum_path(&path) {
         // First try to find a stored checksum file
         let checksum_storage_key = format!("maven/{}", path);
-        let storage = state.storage_for_repo(&repo.storage_path);
         if let Ok(content) = storage.get(&checksum_storage_key).await {
             return Ok(Response::builder()
                 .status(StatusCode::OK)

